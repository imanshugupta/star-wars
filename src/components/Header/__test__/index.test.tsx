import { render, screen } from "@testing-library/react";
import { Header } from "..";

describe("<Header />", () => {
  test("renders the Star Wars logo", () => {
    render(<Header />);
    const logo = screen.getByAltText("logo");
    expect(logo).toBeInTheDocument();
    expect(logo).toHaveAttribute("src", "/star-wars-logo.png");
    expect(logo).toHaveAttribute("height", "100");
  });

  test("match the snapshot", () => {
    const { asFragment } = render(<Header />);
    expect(asFragment()).toMatchSnapshot();
  });
});
