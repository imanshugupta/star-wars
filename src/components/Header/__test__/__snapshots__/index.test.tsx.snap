// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="container-fluid border-bottom"
  >
    <div
      class="container "
    >
      <div
        class="py-3 d-flex"
      >
        <div>
          <img
            alt="logo"
            height="100"
            src="/star-wars-logo.png"
          />
        </div>
        <div
          class="mt-auto ms-3"
        >
          <h1>
            <i>
              Characters
            </i>
          </h1>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
